sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeViewModel.kt

```diff
@@ -12,26 +12,39 @@ import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
-    private val useCase: UseCase,
+    private val getModelsUseCase: GetModelsUseCase,
+    private val getFirstTimeLaunchPreferencesUseCase: GetFirstTimeLaunchPreferencesUseCase,
+    private val updateFirstTimeLaunchPreferencesUseCase: UpdateFirstTimeLaunchPreferencesUseCase,
     dispatchers: DispatchersProvider
 ) : BaseViewModel(dispatchers) {
 
     private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
     val uiModels: StateFlow<List<UiModel>>
         get() = _uiModels
 
+    private val _firstTimeLaunch = MutableStateFlow(false)
+    val firstTimeLaunch: StateFlow<Boolean>
+        get() = _firstTimeLaunch
+
     init {
         execute {
-            showLoading()
-            useCase()
-                .catch {
-                    _error.emit(it)
-                }
-                .collect { result ->
-                    val uiModels = result.toUiModels()
-                    _uiModels.emit(uiModels)
+            val getModelsFlow = getModelsUseCase()
+            val getFirstTimeLaunchPreferencesFlow = getFirstTimeLaunchPreferencesUseCase()
+
+            getModelsFlow.combine(getFirstTimeLaunchPreferencesFlow) { uiModels, firstTimeLaunch ->
+                _uiModels.emit(uiModels.toUiModels())
```

Comment:
Quick question: Why are we emitting new value like this manually? What is the difference between
```
_uiModels.emit(uiModels.toUiModels())
```
and
```
_uiModels.value = uiModels.toUiModels()
```

Reply:
@kaungkhantsoe It's the same. 😁 
https://stackoverflow.com/a/66673965
I can see your PR is using `_navigate.emit` too. https://github.com/nimblehq/android-templates/pull/390/files#diff-4f89083574fd3be2e7422e65bfd3d0f0ae5e5c2321733e28a66a82c4fc44605fR39. 😂

Reply:
@lydiasama I didn't notice it when I was doing my PR. I just edited `buildDestination` to `createRoute` 😬 . It is alright. I just asked because I am curious. Both works.